pkg/sql/schemachanger/scgraph/graph.go

```diff
@@ -79,8 +80,8 @@ func New(initial scpb.State) (*Graph, error) {
 	g := Graph{
 		targetIdxMap:     map[*scpb.Target]int{},
 		nodeOpEdgesFrom:  map[*scpb.Node]*OpEdge{},
-		nodeDepEdgesFrom: map[*scpb.Node][]*DepEdge{},
-		nodeDepEdgesTo:   map[*scpb.Node][]*DepEdge{},
+		nodeDepEdgesFrom: map[*scpb.Node]*btree.BTree{},
+		nodeDepEdgesTo:   map[*scpb.Node]*btree.BTree{},
```

Comment:
my suggestion was that you get rid of the map altogether. You can iterate the prefix of the tree corresponding to this node.